docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,404 @@
+.. _custom_core:
+
+Adding a custom core
+====================
+
+You may want to add a custom RISC-V core to Chipyard generator. If the top module of your core is not in Chisel, 
+you will first need to create a Verilog blackbox for it. See :ref:`incorporating-verilog-blocks` for instructions.
+Once you have a top module in Chisel, you are ready to create integrate it with Chipyard. 
+
+.. note:: 
+
+    RoCC is not supported by custom core currently. Please use Rocket or Boom if you need to use RoCC.
+
+Parameter Case Classes
+----------------------
+
+Chipyard will generate a core for every ``TileParams`` object it discovered in the current config.
+``TileParams`` is a trait containing the information needed to create a tile, and every custom core must implement
+their own version of ``TileParams``, as well as ``CoreParams`` which is passed as a field in ``TileParams``.
+
+``TileParams`` holds the parameters that are the same for every generated core, while ``CoreParams`` contains those
+that can vary from cores to cores. They must be implemented as case classes with fields that can be overridden by 
+other config fragments as the constructor parameters. See the appendix at the bottom of the page for a list of 
+variable to be implemented. You can also add custom fields to them, but standard fields should always be preferred. 
+
+Now you have your parameter classes, you will need config keys to hold them. There are two required keys:
+
+.. code-block:: scala
+
+    case object MyTilesKey extends Field[Seq[MyTileParams]](Nil)
+    case object MyCrossingKey extends Field[Seq[RocketCrossingParams]](List(RocketCrossingParams())) 
+
+``MyCrossingKey`` here is used to store information about the clock-crossing behavior of the core, and it is normally
+set to its default values. 
+
+``TileParams`` and ``CoreParams`` contains the following fields (you may ignore any fields marked "Rocket specific" and 
+use their default values, although it is recommended to use them if you need a custom field with similar purposes) :
+
```

Comment:
This would be a good place to link to the places in BOOM and Ariane which demonstrate how this is used.
In the future, Sodor as well.

Reply:
I added the path to the files at the beginning of the page so that people can follow them when reading this page. Should I use a github link?

Reply:
This is fine